src/status_im/navigation/events.cljs

```diff
@@ -26,21 +26,26 @@
 
 (rf/defn navigate-to-within-stack
   {:events [:navigate-to-within-stack]}
-  [_ comp-id]
-  {:navigate-to-within-stack comp-id})
+  [{:keys [db]} comp-id]
+  (let [modal-view-ids (not-empty (:modal-view-ids db))]
+    (merge {:navigate-to-within-stack comp-id}
+           (when modal-view-ids
+             {:db (assoc db :modal-view-ids (conj modal-view-ids (first comp-id)))}))))
```

Comment:
Thanks for the suggestion @ulisesmac! I'll implement this refactor, looks much cleaner 💯 

> btw, why do we only conj when the modal-views-ids is not empty?

`:navigate-to-within-stack` is likely to be used when navigating within a modal stack, but it can happen that the same screen could be inside a modal stack _or_ inside the root stack, and still `:navigate-to-within-stack` can work for both situations. In the last case, there is no need to update `:modal-view-ids` because we would be just navigating within the root stack, that's why we do the check.

Reply:
thanks for the clarification!